### PR TITLE
Merge text1 and text2 and put const/ewram data in C file

### PR DESCRIPTION
**Pull request by DizzyEggg**

### CODE REVIEW

File: src/text.c [hunk 1]
```diff
@@ -1,48 +1,309 @@
...
+static bool8 xxx_update_bg_vram(UnkTextStruct1 *a0);
+static void sub_800898C(void);
+static void sub_80089AC(const UnkTextStruct2 *r4, UnkTextStruct2_sub *r5_Str);
+static void PrepareTextbox_8008C6C(UnkTextStruct1 *strArr, u32 strId);
+static void xxx_draw_string(UnkTextStruct1 *strArr, s32 x, s32 y, const u8 *str, u32 windowId, u32 terminatingChr, s32 characterSpacing, s32 lineSpacing);
+static const u8 *HandleTextFormat(UnkTextStruct1 *strArr, const u8 *str, struct UnkDrawStringStruct *sp);
+static void sub_8009388(void);
+static s32 InterpretColorChar(u8 a0);
+
+UNUSED static const char sPksDir0[] = "pksdir0";
```
Comment (SethBarberee) [inline src/text.c:72]: This is mainly handled with the `globaldata.h`. Two of them here almost implies another file split we are missing based on information from @Kermalis.

File: src/code_801EE10.c [hunk 2]
```diff
@@ -3,8 +3,9 @@
 #include "constants/input.h"
 #include "input.h"
 #include "items.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_801EE10.c:6]: Triple `text.h` include

File: src/code_8015080.c [hunk 3]
```diff
@@ -4,8 +4,8 @@
 #include "input.h"
 #include "memory.h"
 #include "menu_input.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_8015080.c:7]: A few double `text.h` includes that need to be cleaned up.

**Review by SethBarberee — CHANGES_REQUESTED** [review]

File: src/text.c (hunk 1 above)
Comment (DizzyEggg) [reply src/text.c:72]: These are one after the other, so not sure any file is missing. Could just be something with how these pksdir things were generated.
Looking at const and ewram data file splitting makes little sense here imo

File: src/code_8015080.c (hunk 3 above)
Comment (DizzyEggg) [reply src/code_8015080.c:7]: I fixed these two. Are there any in other files left?

File: src/text.c (hunk 1 above)
Comment (DizzyEggg) [reply src/text.c:72]: I could put one in `random.c` which is just before `text.c`

File: src/code_801B3C0.c [hunk 4]
```diff
@@ -8,8 +8,8 @@
 #include "memory.h"
 #include "menu_input.h"
 #include "sprite.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_801B3C0.c:11]: One here

File: src/code_801B60C.c [hunk 5]
```diff
@@ -14,8 +14,8 @@
 #include "memory.h"
 #include "menu_input.h"
 #include "pokemon.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_801B60C.c:17]: One here.

File: src/code_801C8C4.c [hunk 6]
```diff
@@ -6,8 +6,8 @@
 #include "items.h"
 #include "memory.h"
 #include "menu_input.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_801C8C4.c:9]: One here.

File: src/code_801D014.c [hunk 7]
```diff
@@ -24,8 +24,8 @@
 #include "options_menu1.h"
 #include "party_list_menu.h"
 #include "rescue_team_info.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_801D014.c:27]: One here.

File: src/code_801D9E4.c [hunk 8]
```diff
@@ -7,8 +7,8 @@
 #include "input.h"
 #include "memory.h"
 #include "menu_input.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_801D9E4.c:10]: One here.

File: src/code_801F428.c [hunk 9]
```diff
@@ -7,8 +7,8 @@
 #include "menu_input.h"
 #include "moves.h"
 #include "pokemon.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_801F428.c:10]: One here.

File: src/code_801F808.c [hunk 10]
```diff
@@ -9,8 +9,8 @@
 #include "sprite.h"
 #include "structs/menu.h"
 #include "structs/str_text.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_801F808.c:12]: One here.

File: src/code_80227B8.c [hunk 11]
```diff
@@ -14,8 +14,8 @@
 #include "memory.h"
 #include "menu_input.h"
 #include "pokemon_3.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_80227B8.c:17]: One here.

File: src/code_8023144.c [hunk 12]
```diff
@@ -11,8 +11,8 @@
 #include "memory.h"
 #include "menu_input.h"
 #include "pokemon_3.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_8023144.c:14]: One here.

File: src/code_8024458.c [hunk 13]
```diff
@@ -9,8 +9,8 @@
 #include "menu_input.h"
 #include "pokemon.h"
 #include "pokemon_3.h"
-#include "text1.h"
-#include "text2.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_8024458.c:12]: One here.

File: src/code_8272724.c [hunk 14]
```diff
@@ -1,5 +1,6 @@
 #include "global.h"
-#include "text1.h"
+#include "text.h"
```
Comment (SethBarberee) [inline src/code_8272724.c:2]: One here.

**Review by SethBarberee — CHANGES_REQUESTED** [review]

File: src/text.c (hunk 1 above)
Comment (SethBarberee) [reply src/text.c:72]: Sure, that would work

**Comment by DizzyEggg** [comment]
All should be covered.

**Review by SethBarberee — CHANGES_REQUESTED** [review]
Missed some extra includes in `src/code_801B60C.c`, `src/code_801D9E4.c`, and `src/code_8024458.c`

**Comment by DizzyEggg** [comment]
> Missed some extra includes in `src/code_801B60C.c`, `src/code_801D9E4.c`, and `src/code_8024458.c`

Done

**Review by SethBarberee — APPROVED** [review]

**Pull request merged**